packages/cli/src/lib/run/log-displayer.ts

```diff
@@ -85,16 +67,46 @@ async function logDisplayer(heroku: APIClient, options: LogDisplayerOptions) {
     }
   })
 
-  const response: { body: { logplex_url: string}} = await heroku.post(`/apps/${options.app}/log-sessions`, {
-    body: {
-      tail: options.tail,
+  const firApp = await isFirApp(options.app, heroku)
+  const isTail = firApp || options.tail
+
+  const requestBodyParameters = {
+    source: options.source,
+  }
+
+  if (firApp)
+    Object.assign(requestBodyParameters, {
       dyno: options.dyno,
-      source: options.source,
+      type: options.type,
+    })
+  else
+    Object.assign(requestBodyParameters, {
+      dyno: options.dyno || options.type,
       lines: options.lines,
-    },
-  })
+      tail: options.tail,
+    })
 
-  return readLogs(response.body.logplex_url)
+  let recreateLogSession = false
+  do {
+    const {body: logSession} = await heroku.post<LogSession>(`/apps/${options.app}/log-sessions`, {
+      body: requestBodyParameters,
+      headers: {Accept: 'application/vnd.heroku+json; version=3.sdk'},
+    })
+
+    try {
+      await readLogs(
+        logSession.logplex_url,
+        isTail,
+        firApp ? Number(process.env.HEROKU_LOG_STREAM_TIMEOUT || '15') * 60 * 1000 : undefined,
```

Comment:
Yes, exactly that, it re-sends the request for a new log session so that the stream continues. I tested and the behavior is that your stream "hangs" after 15 minutes and you receive no more updates, but the connection isn't closed, so you won't notice. In order to maintain the same behavior for Cedar and Fir apps, the logs should continue to stream until the users interrupts, so I thought better than returning an error when 15 minutes elapsed it would be better to create a new log session and continue streaming.